core/lib.ml

```diff
@@ -402,7 +402,7 @@ let env : (string * (located_primitive * Types.datatype * pure)) list = [
 
   "spawnWait",
   (`PFun (fun _ -> assert false),
-   datatype "(() ~> a) ~> a",
+   datatype "(() ~_~> a) ~> a",
```

Comment:
I see. The point is that the two implicit effect variables are distinct, but with `shared_effect_vars` enabled they become the same variable. 

I would use the non-sugared form `() { |_}~> a`